js/components/generate/GenerateForm.js

```diff
@@ -0,0 +1,397 @@
+            }
+          })
+          parentsObj[boolId] = parents
+          childrenObj[boolId] = childs
+          outEdgesObj[boolId] = outEdges
+          inEdgesObj[boolId] = inEdges
+        })
+
+        const edgesStatus = Object.values(edgesJSON).reduce((acc, curr) => {
+          const source = curr.source
+          const target = curr.target
+          let status
+          const isSourceSelected =
+            nodesStatus[source]?.selected || boolsStatus[source] === "active"
+
+          const isTargetSelected =
+            nodesStatus[target]?.selected || boolsStatus[target] === "active"
+
+          const targetStatus = nodesStatus[target]?.status || boolsStatus[target]
+
+          if (!isSourceSelected && targetStatus === "missing") {
+            status = "missing"
+          } else if (!isSourceSelected) {
+            status = "inactive"
+          } else if (!isTargetSelected) {
+            status = "takeable"
+          } else {
+            status = "active"
+          }
+          acc[curr.id_] = status
+          return acc
+        }, {})
+
+        this.graph.current.setState({
+          labelsJSON: labelsJSON,
+          regionsJSON: regionsJSON,
+          nodesJSON: nodesJSON,
+          hybridsJSON: hybridsJSON,
+          boolsJSON: boolsJSON,
+          edgesJSON: edgesJSON,
+          nodesStatus: nodesStatus,
+          edgesStatus: edgesStatus,
+          boolsStatus: boolsStatus,
+          width: data.width,
+          height: data.height,
+          zoomFactor: 1,
+          horizontalPanFactor: 0,
+          verticalPanFactor: 0,
+          connections: {
+            parents: parentsObj,
+            inEdges: inEdgesObj,
+            children: childrenObj,
+            outEdges: outEdgesObj,
+          },
+          selectedNodes: storedNodes,
+        })
+      })
+      .catch(err => {
+        console.log("err :>> ", err)
+      })
+  }
+
+  /**
+   * Produce an appropriate warning message string in the case that no courses have been entered
+   * or that one or more invalid courses have been entered.
+   * @param {string} coursesString - The string version of the inputted courses
+   * @param {string[]} invalidCourses - The array of invalid course codes
+   * @returns {string} The warning message string.
+   */
+  computeMessage(coursesString, invalidCourses) {
+    if (!coursesString.length) {
+      return "Cannot generate graph – no courses entered!"
+    } else if (invalidCourses.length === 0) {
+      return ""
+    } else if (invalidCourses.length === 1) {
+      return `The course ${invalidCourses} was invalid! Please check your input.`
+    } else {
+      return `The courses [${invalidCourses}] were invalid! Please check your input.`
+    }
+  }
+
+  render() {
+    return (
+      <div style={{ display: "flex", flexDirection: "row", height: "100%" }}>
+        {/* {console.log(this.state.showWarning, "~"+this.state.courses+"~", this.state.invalidCourses)} */}
```

Comment:
This should be removed. This is one of the things you should be looking for during your self-review of your changes.